libraries/AP_Math/vector3.h

```diff
@@ -265,6 +265,14 @@ class Vector3
 
     // Shortest distance between point(p) to a point contained in the line segment defined by w1,w2
     static float closest_distance_between_line_and_point(const Vector3<T> &w1, const Vector3<T> &w2, const Vector3<T> &p);
+
+    // Point in the line segment defined by w1,w2 which is closest to point(p)
+    static Vector3<T> point_on_line_closest_to_other_point(const Vector3<T> &w1, const Vector3<T> &w2, const Vector3<T> &p);
+
+    // This implementation is borrowed from: http://geomalgorithms.com/a07-_distance.html
```

Comment:
Just to match the other methods, I'd change the first line of the comments to be just, "Shortest distance between two line segments".  I'd then make "INPUT" "Input:" and "OUTPUT:" "Output:".